### PR TITLE
PHPLIB-659 Support options for time-series collections

### PR DESCRIPTION
**Pull request by alcaeus**
PHPLIB-659

### CODE REVIEW

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (alcaeus) [inline docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:142]: Note: I haven't found additional documentation for this param publicly yet, so I've only stubbed out documentation. I suggest we revisit this closer to the 5.0 release cycle if at all. In that case, I'd create a follow-up ticket, but I'm also fine linking to the command documentation for more information.

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (jmikola) [inline docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:137]: I realize this PR pertains to [DRIVERS-1438](https://jira.mongodb.org/browse/DRIVERS-1438), which only added a `timeseries` option, but there is a follow-up ticket ([DRIVERS-1774](https://jira.mongodb.org/browse/DRIVERS-1774)) that moves `expireAfterSeconds` to a top-level option. That spec PR is still being reviewed, so there are not yet any language tickets.

I'd suggest holding off on this PR for the time being, since DRIVERS-1774 will require changing this documentation, `CreateCollection.php`, and resyncing spec tests.

Comment (alcaeus) [reply docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:137]: Ah, thanks for the heads up. I'll give the tests from that PR a quick check to provide feedback and will wait until DRIVERS-1774 is ready before merging.

**Comment by alcaeus** [comment]
Updated to incorporate changes from DRIVERS-1774 and DRIVERS-1779. Note that the docs are very vague at this point, but I was not able to find any public documentation for these fields yet. Should we create a ticket to follow up on documentation before the release of 1.9?

**Comment by jmikola** [comment]
> Should we create a ticket to follow up on documentation before the release of 1.9?

Yes. I'd suggest talking to @agolin95 about that in case a separate DRIVERS ticket would be useful. We can then have that block on a relevant ticket for the server manual itself. In this case, I don't think we really need to document the internal structure of `timeseries` in our documentation, but it'd be nice to have an actual URL in the server manual to link to for reference.

**Comment by alcaeus** [comment]
@jmikola rebased on top of master and ensured that the spec test is up to date. Ready for a final review.

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (jmikola) [inline docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:41]: Based on existing docs, this should be phrased as:

> This option is available in MongoDB 5.0+ and will result in an exception at execution time if specified for an older server version.

Alternatively (as is done later in this file for `writeConcern`):

> This is not supported for server versions prior to 5.0 and will result in an exception at execution time if used.

In both cases, these lines are typically added _after_ any documentation describing the option's functionality. For instance, the `writeConcern` docs later in this file add that using `post:`.

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (jmikola) [inline docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:154]: Same comment as `expireAfterSeconds`.

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (jmikola) [inline docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:161]: Same comment as `expireAfterSeconds`.

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (jmikola) [inline docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml:49]: `post:` is generally only helpful if we're inheriting common docs. In this case, it'd be more consistent with existing docs to just append this after two line breaks at the end of `description`.

File: src/Operation/CreateCollection.php
Comment (jmikola) [inline src/Operation/CreateCollection.php:182]: Reading this again with fresh eyes, I find it very odd that the server team didn't use camelCase for this option name (vs. `expireAfterSeconds`).

**Review by jmikola — APPROVED** [review]
LGTM with my suggestions for doc changes.

File: docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml
Comment (alcaeus) [reply docs/includes/apiargs-MongoDBDatabase-method-createCollection-option.yaml]: I've updated the documentation as suggested. Thanks!

**Comment by alcaeus** [comment]
Build failures are addressed in #835, merging despite failing CI.

**Pull request merged**
